src/picongpu/include/particles/Particles.kernel

```diff
@@ -517,13 +518,13 @@ struct PushParticlePerFrame
          * y=1 for dir = 1
          * y=2 for dir = -1
          */
-        const int direction = 1 +
-            (dir.x() + int(dir.x() < 0)*3) +
-            (dir.y() + int(dir.y() < 0)*3) * BOTTOM
-#if (SIMDIM==DIM3)
-            + (dir.z() + int(dir.z() < 0)*3) * BACK
-#endif
-            ;
+        int direction = 1;
+        uint32_t exchangeType = 1; // see libPMacc/include/types.h for RIGHT, BOTTOM and BACK
+        for (uint32_t i = 0; i < simDim; ++i)
+        {
+            direction += (dir[i] + (dir[i] < 0 ? 3 : 0)) * exchangeType; 
+            exchangeType *= 3; // =3^i
+        }
```

Comment:
Me too. Any idea how to use the *magic numbers* from [`libPMacc/include/types.h`](https://github.com/ComputationalRadiationPhysics/picongpu/blob/f0051b7f1c33a716fcee8cf91657f78b96e0bffd/src/libPMacc/include/types.h#L90-L93) in an iterate-able way?